core/primitives-core/src/hash.rs

```diff
@@ -15,6 +15,12 @@ pub struct Digest(pub [u8; 32]);
 #[as_ref(forward)]
 pub struct CryptoHash(pub Digest);
 
+impl CryptoHash {
+    pub fn raw_bytes(&self) -> &[u8; 32] {
+        &self.0 .0
+    }
+}
```

Comment:
There is already `AsRef` implemented for `CryptoHash`, just use `x.as_ref()` and drop this impl.

Reply:
Not possible, as some use cases need exactly sized array, i.e. `SerializationError { hash: [u8; 32] }` and `as_ref` returns `&[u8]`. It may be worth reworking, but looks like something not belonging to this PR.